src/compile_cache.cc

```diff
@@ -223,13 +226,93 @@ void CompileCacheHandler::ReadCacheFile(CompileCacheEntry* entry) {
   Debug(" success, size=%d\n", total_read);
 }
 
+#ifdef _WIN32
+constexpr bool IsWindowsDeviceRoot(const char c) noexcept {
+  return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
+}
+#endif
+
+static std::string NormalisePath(std::string_view path) {
+  std::string normalised_string(path);
```

Comment:
It would be helpful to document what a "normalized path" is.

Also, super nit, but we normally use en-US spelling for things, which would make this `NormalizePath(...)`, but consider that extremely optional ;-)